docs/Superoperator representations.md

```diff
@@ -12,49 +16,565 @@ where $i$ is a row and $j$ is a column index.
+
+$\begin{align}
+\mathcal C &= I\otimes \mathcal E (|\eta \rangle \langle \eta|) \\\\
+&=\sum_{ij} |i\rangle \langle j| \otimes  \mathcal E (|i \rangle \langle j | ).
 \end{align}$
+
+## Process or $\chi$ matrix to Pauli transfer matrix
+
+$$(R_{\mathcal E})_{i,j} = \frac 1 d \sum_{k,l}\chi_{k,l} {\rm Tr}[ P_i P_k P_j P_l].$$
+
+
+## Superoperator to Pauli transfer matrix
+The standard basis on $n$ qubits is called the computational basis. It is essentially all the strings $|c_1\rangle=|0..0\rangle$ through to $|c_{\rm max}\rangle = |1...1\rangle$. To convert between a superoperator and Pauli transfer matrix  representation we need to to a change of basis from the computational basis to the  Pauli basis. This is acheived by the unitary
+
+$$ U_{c2p}= \sum_{k=1}|c_k\rangle\langle\langle P_k|.$$
+
+The we have
+
+$$ R_{\mathcal E} =  U_{c2p} \mathcal E U_{c2p}^\dagger.$$
+
+
+## Superoperator to Choi
+The conversion from the superoperator to a Choi matrix $\mathcal C$ is simply a (column) reshuffling operation
+
+$$ \mathcal C = R(\mathcal E).$$
+
+It turns out that $ \mathcal E = R(\mathcal C)$ which means that $\mathcal E= R(R(\mathcal E))$.
+
+## Pauli transfer matrix to Superoperator
+To convert between the Pauli transfer matrix and the superoperator representation we need to to a change of basis from the Pauli basis to the computational basis. This is acheived by the unitary
+
+$$ U_{p2c}= \sum_{k=1}|P_k\rangle\rangle \langle k|,$$
+which is simply $U_{c2p}^\dagger$.
+
+The we have
+
+$$\mathcal E =  U_{p2c}R_{\mathcal E}U_{p2c}^\dagger.$$
+
+
+## Pauli transfer matrix to Choi
+We obtain the normalized Choi matrix using the expression
+
+$$ \rho_{\mathcal E} = \frac{1}{d^2}\sum_{i,j=1}^{d^2} (R_{\mathcal E})_{i,j}  \, P_j^T \otimes P_i.$$
+
+
+
+## Choi to Kraus
+This is simply the reverse of the Kraus to Choi procedure.
+
+Given the Choi matrix $\mathcal C$ we find it's eigenvalues $\{\lambda_i\}$ and vectors $\{|M_i\rangle\rangle \}$. Then the Kraus operators are
+
+$$ M_i = \sqrt{\lambda_i}\, {\rm unvec}\big (|M_i\rangle\rangle\big),$$
+
+For numerical implementation one usually puts a threshold on the eigenvalues, say $\lambda> 10^{-10}$, to prevent numerical instablities. 
+
+## Choi to Pauli transfer matrix
+First we normalize the Choi representation
+
+$$\begin{align}
+\rho_{\mathcal E}=\frac 1 d \mathcal C = \frac 1 d \sum_{ij} |i\rangle \langle j| \otimes  \mathcal E (|i \rangle \langle j | )
+\end{align}$$ 
+
+Then the matrix elements of the Pauli transfer matrix representation of the channel can be obtained from the Choi state using
+
+$$(R_{\mathcal E})_{i,j} ={\rm Tr}[ \rho_{\mathcal E} \, P_j^T \otimes P_i].$$
+
+## Choi to Superoperator
+The conversion from a Choi matrix $\mathcal C$  to a superoperator is simply a (column) reshuffling operation
+
+$$ \mathcal E = R(\mathcal C).$$
+
+It turns out that $ \mathcal C = R(\mathcal E)$ which means that $\mathcal C= R(R(\mathcal C))$.  
+
+  
+
+## Examples: One qubit channels
```

Comment:
Nice observations to add: 

- the Choi matrix of a unitary process always has rank 1.  

- The Liouville representation of a unitary process is always full rank. 

- The eigenvalues of a Choi matrix give you an upper bound to the probability a particular (canonical) Kraus operator will occur (generally that probability depends on the state).  This is helpful when sampling Kraus operators (you can test for which occurred accoridng to the order of these eigenvalues).

- the chi matrix (in the Pauli basis) is very convenient for computing the result of Pauli twirling or Clifford twirling the corresponding process.

Reply:
done